### PR TITLE
doc: rework installation/setup guides, general refactoring

### PR DESCRIPTION
**Pull request by nodiscc**
A full and concise installation guide with examples is a frequent request.

The documentation should provide such a guide for basic installation needs, while explaining alternative/advanced configuration at the end of teh document.

Links to reference guides and documentation should be used more frequently to avoid recommending an outdated or excessively complex configuration.

Ideally the download, installation, server setup and security sections should fit on a single page.

Another page can walk the user through installing Shaarli in Docker behind a reverse proxy.

- Fixes https://github.com/shaarli/Shaarli/issues/1381
- Fixes https://github.com/shaarli/Shaarli/issues/1455

### CODE REVIEW

**Comment by nodiscc** [comment]
Still work in progress, see:
 - https://shaarli.readthedocs.io/en/doc-rework-setup/Server-configuration/ (currently redoing/condensing the nginx section)
 - https://shaarli.readthedocs.io/en/doc-rework-setup/Download-and-Installation/
 - https://shaarli.readthedocs.io/en/doc-rework-setup/Release-Shaarli/

As you can see this PR also proposes dropping support for the `stable` branch. I think the current model is complex/confusing, it adds maintenance and has no great benefits. Under the proposed model you can:
 - Install the latest tag/[release](https://github.com/shaarli/Shaarli/releases) using any available method (release zip, git clone/zip + composer). Upgrades to the next release must be done manually (overwrite with new zip contents, or `git remote update; git checkout new_tag; composer ...`)
 - Install the `release` branch from git clone/zip (the only benefit `git remote update; git checkout` is simply replaced with `git pull`), or the `master` branch using the same method.

In short the `release` branch would now be considered as `stable` (in its current form it's kind of like an `oldstable` branch.). What do you think?

**Comment by KyleWJohnston** [comment]
In `doc/md/Server-configuration.md`, I had to have the `Directory` section outside of `VirtualHost` rather than within it, but it may be my configuration (see #1423). I also had to change the configuration within `Directory` as a result of running a newer version of Apache. See http://httpd.apache.org/docs/current/upgrading.html.

I'm running Arch Linux and have only had my webserver for a few months. If it works for you, feel free to ignore.

**Comment by nodiscc** [comment]
I will definitely re-test all setup procedures before removing the work-in-progress status. Apache 2.4 will be used for the test. [\<Directory>](https://httpd.apache.org/docs/2.4/mod/core.html#directory) should work inside a virtualhost definition but I will double-check everything, thanks

**Comment by nodiscc** [comment]
Rebased, squashed.

- See the [commit message](https://github.com/shaarli/Shaarli/pull/1389/commits/2fec3f836183797e746b9354a788a30c3c9a684b) for details
- See the [diff](https://github.com/shaarli/Shaarli/pull/1389/files)
- Seethe [rendered documentation](https://shaarli.readthedocs.io/en/doc-rework-setup/)
- Feel free to ask about anything

I still have to test some procedures and change ordering (for example I know the setup procedure still takes you back and forth between 2 pages), but the bulk of it is done. Let me know what you think.

File: doc/md/Community-and-related-software.md [hunk 1]
```diff
@@ -1,54 +1,68 @@
...
 - [shaarli2mastodon](https://github.com/kalvn/shaarli2mastodon) by [@kalvn](https://github.com/kalvn) - This Shaarli plugin allows you to automatically publish links you post on your Mastodon timeline.
-- [shaarli-descriptor](https://github.com/immanuelfodor/shaarli-descriptor) by [@immanuelfodor](https://github.com/immanuelfodor) - Customize the default height/number of rows of the Description field when editing a link.
+- [shaarli-descriptor](https://github.com/immanuelfodor/shaarli-descriptor) by [@immanuelfodor](https://github.com/immanuelfodor) - Customize the default height/number of rows of the Description field when editing a Shaare.
 - [urlextern](https://github.com/trailjeep/shaarli-urlextern) by [@trailjeep](https://github.com/trailjeep) - Shaarli plugin to open external links in a new tab/window.
-- [favicons](https://github.com/trailjeep/shaarli-favicons) by [@trailjeep](https://github.com/trailjeep) - Shaarli plugin to add favicon/filetype icons to links.
+- [favicons](https://github.com/trailjeep/shaarli-favicons) by [@trailjeep](https://github.com/trailjeep) - Shaarli plugin to add favicon/filetype icons to Shaares.
```
Comment (ArthurHoaro) [inline doc/md/Community-and-related-software.md:27]: We should fix the alphabetical sort for these.

File: doc/md/Plugins.md [hunk 2]
```diff
@@ -55,25 +54,25 @@ $GLOBALS['config']['ENABLED_PLUGINS'] = array(
 );
 ```
```
Comment (ArthurHoaro) [inline doc/md/Plugins.md:55]: This is outdated.

File: doc/md/Troubleshooting.md [hunk 3]
```diff
@@ -8,38 +11,48 @@ Delete the file `data/config.json.php` and display the page again. You will be a
...
+
 ## Hosting problems
 
 ### Old PHP versions
 
-On **free.fr**: free.fr now supports php 5.6.x([link](http://les.pages.perso.chez.free.fr/migrations/php5v6.io))
-and so support now the tag autocompletion but you have to do the following.
-
-At the root of your webspace create a `sessions` directory and a `.htaccess` file containing:
+- On hosts (such as **free.fr**) which only support PHP 5.6, Shaarli [v0.10.4](https://github.com/shaarli/Shaarli/releases/tag/v0.10.4) is the maximum supported version. At the root of your webspace create a `sessions` directory and a `.htaccess` file containing:
```
Comment (agentcobra) [inline doc/md/Troubleshooting.md:41]:
```suggestion
 - On hosts (such as **free.fr**) which only support PHP 5.6, Shaarli [v0.10.4](https://github.com/shaarli/Shaarli/releases/tag/v0.10.4) is the maximum supported version. At the root of your webspace create a `sessions` directory and a `.htaccess` file containing:
```

**Comment by nodiscc** [comment]
I solved my [NAT problems](https://github.com/firehol/firehol/issues/432) which were preventing me from testing install procedures with proper Let's Encrypt certificate generation. I will do another pass this weekend and probably include some [asccinema](https://asciinema.org/) demos of the process.

**Comment by Lucas-C** [comment]
Wow, great work you did there !
Thanks !

**Comment by nodiscc** [comment]
Note for self:

- [ ] re-test installation from source
- [ ] re-test installation from release ZIP
- [ ] check apache configuration
- [ ] check nginx configuration
- [ ] re-test installation from Docker

Once this is done, I think it will be good to merge.

There will be conflicts with https://github.com/shaarli/Shaarli/pull/1511 though. @ArthurHoaro How will we do this? I can move your documentation changes to this PR - it could be less tedious than resolving merge conflicts after the fact.

**Comment by ArthurHoaro** [comment]
I don't have many changes regarding the documentation. I can merge #1511 and resolve the conflicts here myself. 

Do you want any help with the server configuration tests?

**Comment by ArthurHoaro** [comment]
I have rebased the branch from master and force pushed it. There weren't a lot of conflicts so I think we're fine but I have a local save just in case.

If you want to you can rebase+force push to take back ownership of all these commits.

**Comment by ArthurHoaro** [comment]
> Didn't have time to recheck the full docker setup (shaarli.readthedocs.io/en/doc-rework-setup/Docker) - because I am still shaving yak and tweaking things on my docker hosts.... if you want to test it and add whatever you find missing, that would be great.

I made a few changes and tested the docker install + run Shaarli image in a fresh VM, everything is working fine. I didn't test the Traefik one because it requires remote access and proper DNS configuration, which I don't have/would require quite some time to setup locally.

> Apache/nginx/cerbot-based setups seem to work fine, I added a short screencast.

That's GREAT! Thank you for taking the time to do that! :+1:

**Comment by nodiscc** [comment]
rebased.
